client/src/document/hooks.ts

```diff
@@ -8,3 +11,95 @@ export function useDocumentURL() {
   // pathname with the document's mdn_url.
   return url.endsWith("/") ? url.substring(0, url.length - 1) : url;
 }
+
+export function useCopyExamplesToClipboard(doc: Doc | undefined) {
+  React.useEffect(() => {
+    if (!doc) {
+      return;
+    }
+
+    if (!isClipboardSupported()) {
+      console.log(
+        "Copy-to-clipboard disabled because your browser does not appear to support it"
+      );
+      // bail and don't inject buttons to DOM
+      return;
+    }
+
+    [...document.querySelectorAll("div.code-example pre")].forEach(
+      (element) => {
+        const userMessage = document.createElement("span");
+        const button = document.createElement("button");
+        const span = document.createElement("span");
+        userMessage.textContent = "Copied!";
+        span.textContent = "Copy to Clipboard";
+
+        const wrapper: HTMLElement | null = element.parentElement;
+
+        userMessage.setAttribute("class", "user-message");
+        userMessage.setAttribute("aria-hidden", "true");
+        button.setAttribute("aria-hidden", "false");
+        button.setAttribute("type", "button");
+        button.setAttribute("class", "copy-icon");
+        span.setAttribute("class", "visually-hidden");
+
+        button.appendChild(span);
+        wrapper?.appendChild(button);
+        wrapper?.appendChild(userMessage);
+
+        button.onclick = () => {
+          const copiedSuccessfully = copyToClipboard(element);
```

Comment:
> But by the way, what happened to the code that uses to make an exception for Internet Explorer?

It was removed since the early return was added in the custom hook if the Clipboard Web API isn't supported

```js
if (!isClipboardSupported()) {
      console.log(
        "Copy-to-clipboard disabled because your browser does not appear to support it"
      );
      // bail and don't inject buttons to DOM
      return;
    }
```

instead of returning early, we should use `window.clipboardData.setData()` to provide an option for IE users to copy-to-clipboard, if that were to fail (which I need to go look how it would) then we could log "Copy-to-clipboard disabled because your browser does not appear to support it"

Reply:
@peterbe On line 56 in your suggested change, in the catch block did you mean for `copiedSuccessfully` to equal `false`? Just wanted to double check before pushing commits